src/Machete.HL7.Tests/AdvancedQueryTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿namespace Machete.HL7.Tests
+{
+    using HL7Schema.V26;
+    using NUnit.Framework;
+    using Testing;
+
+
+    [TestFixture]
+    public class AdvancedQueryTests :
+        MacheteHL7TestContext<Version26>
+    {
+        [Test]
+        public void Should_be_able_to_query_blocks()
+        {
+            const string message = @"MSH|^~\&|NextGen|^DOSC|RH|18779|20130405125146269||ORM^O01|1999077678|P|2.3|||AL|AL
```

Comment:
What is the source of this sample message? It appears to contain data that may be identifiable. Test data should be hand-crafted, original, and not pulled from a production system.